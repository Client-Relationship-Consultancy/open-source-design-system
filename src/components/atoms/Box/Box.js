import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"
import { colourPalette } from "../../../brandColours.ts"

const StyledBox = styled.div`
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  > * {
    padding: 0.5rem 1rem;
  }
  ${props =>
    props.onClick &&
    `
    :hover {
      cursor: pointer;
    }
  `}
`
StyledBox.displayName = "StyledBox"

const Title = styled.div`
  background-color: ${props => props.theme.primary.main.hex};
  width: 100%;
  border-radius: 10px 10px 0 0;
  > h4 {
    margin: 0;
    color: ${props => props.theme.primary.main.on};
    font-size: 1.2rem;
  }
`
Title.defaultProps = {
  theme: colourPalette.examplePalette,
}
Title.displayName = "Title"

const Body = styled.div`
  background-color: ${props => props.theme.surface.hex};
  border-radius: ${props => (props.flatTop ? "0 0 10px 10px" : "10px")};
  border: 1px solid ${props => props.theme.primary.main.hex};
`
Body.defaultProps = {
  theme: colourPalette.examplePalette,
}
Body.displayName = "Body"

const Box = props => {
  const { title, children, id, className, ...other } = props
  return (
    <StyledBox id={id} className={className} {...other}>
      {title && (
        <Title>
          <h4>{title}</h4>
        </Title>
      )}
      <Body flatTop={!!title}>{children}</Body>
    </StyledBox>
  )
}
Box.displayName = "Box"

Box.propTypes = {
  title: PropTypes.any,
  children: PropTypes.node,
  id: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func,
}

export default Box
