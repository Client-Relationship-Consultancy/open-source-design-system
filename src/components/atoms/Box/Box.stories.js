/* eslint no-param-reassign: 0 */

import React from "react"

import { storiesOf } from "@storybook/react"
import { boolean } from "@storybook/addon-knobs"

import { Box, FormCheckbox } from "../../../index"

const stories = storiesOf("Layout", module)

// Edit existing global addons
stories.addParameters({
  info: {
    propTablesExclude: [FormCheckbox], // Exludes FormCheckbox from propTables
  },
})

stories.add("Layout Box Template", context => {
  // Adding properties to story so that the global wrapper can reach it.
  // Acccepts [description, dos, donts]
  // 'description' can be a string or an array for rendering paragraphs
  context.description = "This is the standard wrapper for components on a page."
  // 'dos' accepts an array and renders an ul list
  context.dos = [
    "Use this wrapper to help separate different sections on a page",
    "Only wrap the high level sections of the page",
  ]
  // 'donts' accepts an array and renders an ul list
  context.donts = [
    "Don't wrap every section with this wrapper",
    "Don't Forget to specify the width",
  ]

  return (
    <Box title={boolean("Show Title", true) && "Standard"}>
      <div style={{ display: "flex", flexDirection: "column" }}>
        <FormCheckbox label="Account" name="fields" className="normal" id="account" />
        <FormCheckbox label="Market" name="fields" className="normal" id="market" />
        <FormCheckbox label="DB Contact" name="db-contact" className="normal" id="db-contact" />
      </div>
    </Box>
  )
})
