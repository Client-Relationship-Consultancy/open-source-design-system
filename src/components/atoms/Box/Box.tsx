import React from "react";
import styled from "styled-components";
import { colourPalette } from "../../../brandColours";

const StyledBox = styled.div`
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  > * {
    padding: 0.5rem 1rem;
  }
  ${props =>
    props.onClick &&
    `
    :hover {
      cursor: pointer;
    }
  `}
`;

StyledBox.displayName = "StyledBox";

const Title = styled.div`
  background-color: ${props => props.theme.primary.main.hex};
  width: 100%;
  border-radius: 10px 10px 0 0;
  > h4 {
    margin: 0;
    color: ${props => props.theme.primary.main.on};
    font-size: 1.2rem;
  }
`;

Title.defaultProps = {
  theme: colourPalette.examplePalette,
};

Title.displayName = "Title";

interface IBodyProps {
  flatTop: boolean;
}

const Body = styled.div<IBodyProps>`
  background-color: ${props => props.theme.surface.hex};
  border-radius: ${props => (props.flatTop ? "0 0 10px 10px" : "10px")};
  border: 1px solid ${props => props.theme.primary.main.hex};
`;

Body.defaultProps = {
  theme: colourPalette.examplePalette,
};

Body.displayName = "Body";

type DivProps = React.HTMLProps<HTMLDivElement>;

const Box: React.FC<DivProps> = ({ title, children, id, className, onClick }) => {
  return (
    <StyledBox id={id} className={className} onClick={onClick}>
      {title && (
        <Title>
          <h4>{title}</h4>
        </Title>
      )}
      <Body flatTop={!!title}>{children}</Body>
    </StyledBox>
  )
};

Box.displayName = "Box";

export default Box;
