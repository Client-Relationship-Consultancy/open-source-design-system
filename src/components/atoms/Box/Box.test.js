import React from "react"
import { mount, shallow } from "enzyme"
import "jest-styled-components"
import toJson from "enzyme-to-json"
import Box from "./Box"

describe("Box test", () => {
  const onClick = jest.fn()
  const component = mount(
    <Box title="Title" onClick={onClick}>
      <span />
    </Box>,
  )
  const noTitleComponent = shallow(
    <Box onClick={onClick}>
      <span>heleo</span>
    </Box>,
  )
  it("should match last snapshot with a title", () => {
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it("should call a function when onClick is passed", () => {
    noTitleComponent.simulate("click")
    expect(onClick).toHaveBeenCalled()
  })

  it("should match last snapshot without a title", () => {
    const tree = toJson(noTitleComponent)
    expect(tree).toMatchSnapshot()
  })
})
