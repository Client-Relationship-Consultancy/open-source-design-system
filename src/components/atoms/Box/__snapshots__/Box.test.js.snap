// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box test should match last snapshot with a title 1`] = `
.c0 {
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 > * {
  padding: 0.5rem 1rem;
}

.c0:hover {
  cursor: pointer;
}

.c1 {
  background-color: #A42824;
  width: 100%;
  border-radius: 10px 10px 0 0;
}

.c1 > h4 {
  margin: 0;
  color: #FFFFFF;
  font-size: 1.2rem;
}

.c2 {
  background-color: #FFFFFF;
  border-radius: 0 0 10px 10px;
  border: 1px solid #A42824;
}

<Box
  onClick={[MockFunction]}
  title="Title"
>
  <StyledBox
    onClick={[MockFunction]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Box__StyledBox-sc-1hhoqar-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "border-radius:10px;display:flex;flex-direction:column;justify-content:flex-start;> *{padding:0.5rem 1rem;}",
              [Function],
            ],
          },
          "displayName": "StyledBox",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Box__StyledBox-sc-1hhoqar-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[MockFunction]}
    >
      <div
        className="c0"
        onClick={[MockFunction]}
      >
        <Title
          theme={
            Object {
              "action": Object {
                "dark": Object {
                  "RGB": "79,38,98",
                  "hex": "#4E2662",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "173,101,207",
                  "hex": "#AD65CF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "116,60,143",
                  "hex": "#743C8F",
                  "on": "#FFFFFF",
                },
              },
              "background": Object {
                "RGB": "246,246,246",
                "hex": "#F6F6F6",
              },
              "black": Object {
                "main": Object {
                  "RGB": "39,39,39",
                  "hex": "#272727",
                  "on": "#FFFFFF",
                },
                "tint20": Object {
                  "RGB": "236,236,236",
                  "hex": "#ECECEC",
                  "on": "#272727",
                },
                "tint40": Object {
                  "RGB": "199,199,199",
                  "hex": "#C7C7C7",
                  "on": "#272727",
                },
                "tint60": Object {
                  "RGB": "160,160,160",
                  "hex": "#A0A0A0",
                  "on": "#FFFFFF",
                },
                "tint80": Object {
                  "RGB": "105,105,105",
                  "hex": "#696969",
                  "on": "#FFFFFF",
                },
              },
              "complimentary": Object {
                "dark": Object {
                  "RGB": "206,185,0",
                  "hex": "#CEB900",
                  "on": "#272727",
                },
                "light": Object {
                  "RGB": "255,245,157",
                  "hex": "#FFF59D",
                  "on": "#272727",
                },
                "main": Object {
                  "RGB": "255,238,82",
                  "hex": "#FFEE52",
                  "on": "#272727",
                },
              },
              "error": Object {
                "dark": Object {
                  "RGB": "197,73,1",
                  "hex": "#C54901",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "255,115,32",
                  "hex": "#FF7320",
                  "on": "#FFFFFF",
                },
              },
              "primary": Object {
                "dark": Object {
                  "RGB": "102,22,20",
                  "hex": "#661614",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "218,52,47",
                  "hex": "#DA342F",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "164,40,36",
                  "hex": "#A42824",
                  "on": "#FFFFFF",
                },
              },
              "secondary": Object {
                "dark": Object {
                  "RGB": "5,38,59",
                  "hex": "#05263B",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "89,156,175",
                  "hex": "#599CAF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "11,70,107",
                  "hex": "#0B466B",
                  "on": "#FFFFFF",
                },
              },
              "surface": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
              "white": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
            }
          }
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "_foldedDefaultProps": Object {
                  "theme": Object {
                    "action": Object {
                      "dark": Object {
                        "RGB": "79,38,98",
                        "hex": "#4E2662",
                        "on": "#FFFFFF",
                      },
                      "light": Object {
                        "RGB": "173,101,207",
                        "hex": "#AD65CF",
                        "on": "#FFFFFF",
                      },
                      "main": Object {
                        "RGB": "116,60,143",
                        "hex": "#743C8F",
                        "on": "#FFFFFF",
                      },
                    },
                    "background": Object {
                      "RGB": "246,246,246",
                      "hex": "#F6F6F6",
                    },
                    "black": Object {
                      "main": Object {
                        "RGB": "39,39,39",
                        "hex": "#272727",
                        "on": "#FFFFFF",
                      },
                      "tint20": Object {
                        "RGB": "236,236,236",
                        "hex": "#ECECEC",
                        "on": "#272727",
                      },
                      "tint40": Object {
                        "RGB": "199,199,199",
                        "hex": "#C7C7C7",
                        "on": "#272727",
                      },
                      "tint60": Object {
                        "RGB": "160,160,160",
                        "hex": "#A0A0A0",
                        "on": "#FFFFFF",
                      },
                      "tint80": Object {
                        "RGB": "105,105,105",
                        "hex": "#696969",
                        "on": "#FFFFFF",
                      },
                    },
                    "complimentary": Object {
                      "dark": Object {
                        "RGB": "206,185,0",
                        "hex": "#CEB900",
                        "on": "#272727",
                      },
                      "light": Object {
                        "RGB": "255,245,157",
                        "hex": "#FFF59D",
                        "on": "#272727",
                      },
                      "main": Object {
                        "RGB": "255,238,82",
                        "hex": "#FFEE52",
                        "on": "#272727",
                      },
                    },
                    "error": Object {
                      "dark": Object {
                        "RGB": "197,73,1",
                        "hex": "#C54901",
                        "on": "#FFFFFF",
                      },
                      "main": Object {
                        "RGB": "255,115,32",
                        "hex": "#FF7320",
                        "on": "#FFFFFF",
                      },
                    },
                    "primary": Object {
                      "dark": Object {
                        "RGB": "102,22,20",
                        "hex": "#661614",
                        "on": "#FFFFFF",
                      },
                      "light": Object {
                        "RGB": "218,52,47",
                        "hex": "#DA342F",
                        "on": "#FFFFFF",
                      },
                      "main": Object {
                        "RGB": "164,40,36",
                        "hex": "#A42824",
                        "on": "#FFFFFF",
                      },
                    },
                    "secondary": Object {
                      "dark": Object {
                        "RGB": "5,38,59",
                        "hex": "#05263B",
                        "on": "#FFFFFF",
                      },
                      "light": Object {
                        "RGB": "89,156,175",
                        "hex": "#599CAF",
                        "on": "#FFFFFF",
                      },
                      "main": Object {
                        "RGB": "11,70,107",
                        "hex": "#0B466B",
                        "on": "#FFFFFF",
                      },
                    },
                    "surface": Object {
                      "RGB": "255,255,255",
                      "hex": "#FFFFFF",
                      "on": "#272727",
                    },
                    "white": Object {
                      "RGB": "255,255,255",
                      "hex": "#FFFFFF",
                      "on": "#272727",
                    },
                  },
                },
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Box__Title-sc-1hhoqar-1",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "background-color:",
                    [Function],
                    ";width:100%;border-radius:10px 10px 0 0;> h4{margin:0;color:",
                    [Function],
                    ";font-size:1.2rem;}",
                  ],
                },
                "displayName": "Title",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Box__Title-sc-1hhoqar-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            theme={
              Object {
                "action": Object {
                  "dark": Object {
                    "RGB": "79,38,98",
                    "hex": "#4E2662",
                    "on": "#FFFFFF",
                  },
                  "light": Object {
                    "RGB": "173,101,207",
                    "hex": "#AD65CF",
                    "on": "#FFFFFF",
                  },
                  "main": Object {
                    "RGB": "116,60,143",
                    "hex": "#743C8F",
                    "on": "#FFFFFF",
                  },
                },
                "background": Object {
                  "RGB": "246,246,246",
                  "hex": "#F6F6F6",
                },
                "black": Object {
                  "main": Object {
                    "RGB": "39,39,39",
                    "hex": "#272727",
                    "on": "#FFFFFF",
                  },
                  "tint20": Object {
                    "RGB": "236,236,236",
                    "hex": "#ECECEC",
                    "on": "#272727",
                  },
                  "tint40": Object {
                    "RGB": "199,199,199",
                    "hex": "#C7C7C7",
                    "on": "#272727",
                  },
                  "tint60": Object {
                    "RGB": "160,160,160",
                    "hex": "#A0A0A0",
                    "on": "#FFFFFF",
                  },
                  "tint80": Object {
                    "RGB": "105,105,105",
                    "hex": "#696969",
                    "on": "#FFFFFF",
                  },
                },
                "complimentary": Object {
                  "dark": Object {
                    "RGB": "206,185,0",
                    "hex": "#CEB900",
                    "on": "#272727",
                  },
                  "light": Object {
                    "RGB": "255,245,157",
                    "hex": "#FFF59D",
                    "on": "#272727",
                  },
                  "main": Object {
                    "RGB": "255,238,82",
                    "hex": "#FFEE52",
                    "on": "#272727",
                  },
                },
                "error": Object {
                  "dark": Object {
                    "RGB": "197,73,1",
                    "hex": "#C54901",
                    "on": "#FFFFFF",
                  },
                  "main": Object {
                    "RGB": "255,115,32",
                    "hex": "#FF7320",
                    "on": "#FFFFFF",
                  },
                },
                "primary": Object {
                  "dark": Object {
                    "RGB": "102,22,20",
                    "hex": "#661614",
                    "on": "#FFFFFF",
                  },
                  "light": Object {
                    "RGB": "218,52,47",
                    "hex": "#DA342F",
                    "on": "#FFFFFF",
                  },
                  "main": Object {
                    "RGB": "164,40,36",
                    "hex": "#A42824",
                    "on": "#FFFFFF",
                  },
                },
                "secondary": Object {
                  "dark": Object {
                    "RGB": "5,38,59",
                    "hex": "#05263B",
                    "on": "#FFFFFF",
                  },
                  "light": Object {
                    "RGB": "89,156,175",
                    "hex": "#599CAF",
                    "on": "#FFFFFF",
                  },
                  "main": Object {
                    "RGB": "11,70,107",
                    "hex": "#0B466B",
                    "on": "#FFFFFF",
                  },
                },
                "surface": Object {
                  "RGB": "255,255,255",
                  "hex": "#FFFFFF",
                  "on": "#272727",
                },
                "white": Object {
                  "RGB": "255,255,255",
                  "hex": "#FFFFFF",
                  "on": "#272727",
                },
              }
            }
          >
            <div
              className="c1"
            >
              <h4>
                Title
              </h4>
            </div>
          </StyledComponent>
        </Title>
        <Body
          flatTop={true}
          theme={
            Object {
              "action": Object {
                "dark": Object {
                  "RGB": "79,38,98",
                  "hex": "#4E2662",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "173,101,207",
                  "hex": "#AD65CF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "116,60,143",
                  "hex": "#743C8F",
                  "on": "#FFFFFF",
                },
              },
              "background": Object {
                "RGB": "246,246,246",
                "hex": "#F6F6F6",
              },
              "black": Object {
                "main": Object {
                  "RGB": "39,39,39",
                  "hex": "#272727",
                  "on": "#FFFFFF",
                },
                "tint20": Object {
                  "RGB": "236,236,236",
                  "hex": "#ECECEC",
                  "on": "#272727",
                },
                "tint40": Object {
                  "RGB": "199,199,199",
                  "hex": "#C7C7C7",
                  "on": "#272727",
                },
                "tint60": Object {
                  "RGB": "160,160,160",
                  "hex": "#A0A0A0",
                  "on": "#FFFFFF",
                },
                "tint80": Object {
                  "RGB": "105,105,105",
                  "hex": "#696969",
                  "on": "#FFFFFF",
                },
              },
              "complimentary": Object {
                "dark": Object {
                  "RGB": "206,185,0",
                  "hex": "#CEB900",
                  "on": "#272727",
                },
                "light": Object {
                  "RGB": "255,245,157",
                  "hex": "#FFF59D",
                  "on": "#272727",
                },
                "main": Object {
                  "RGB": "255,238,82",
                  "hex": "#FFEE52",
                  "on": "#272727",
                },
              },
              "error": Object {
                "dark": Object {
                  "RGB": "197,73,1",
                  "hex": "#C54901",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "255,115,32",
                  "hex": "#FF7320",
                  "on": "#FFFFFF",
                },
              },
              "primary": Object {
                "dark": Object {
                  "RGB": "102,22,20",
                  "hex": "#661614",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "218,52,47",
                  "hex": "#DA342F",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "164,40,36",
                  "hex": "#A42824",
                  "on": "#FFFFFF",
                },
              },
              "secondary": Object {
                "dark": Object {
                  "RGB": "5,38,59",
                  "hex": "#05263B",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "89,156,175",
                  "hex": "#599CAF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "11,70,107",
                  "hex": "#0B466B",
                  "on": "#FFFFFF",
                },
              },
              "surface": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
              "white": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
            }
          }
        >
          <StyledComponent
            flatTop={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "_foldedDefaultProps": Object {
                  "theme": Object {
                    "action": Object {
                      "dark": Object {
                        "RGB": "79,38,98",
                        "hex": "#4E2662",
                        "on": "#FFFFFF",
                      },
                      "light": Object {
                        "RGB": "173,101,207",
                        "hex": "#AD65CF",
                        "on": "#FFFFFF",
                      },
                      "main": Object {
                        "RGB": "116,60,143",
                        "hex": "#743C8F",
                        "on": "#FFFFFF",
                      },
                    },
                    "background": Object {
                      "RGB": "246,246,246",
                      "hex": "#F6F6F6",
                    },
                    "black": Object {
                      "main": Object {
                        "RGB": "39,39,39",
                        "hex": "#272727",
                        "on": "#FFFFFF",
                      },
                      "tint20": Object {
                        "RGB": "236,236,236",
                        "hex": "#ECECEC",
                        "on": "#272727",
                      },
                      "tint40": Object {
                        "RGB": "199,199,199",
                        "hex": "#C7C7C7",
                        "on": "#272727",
                      },
                      "tint60": Object {
                        "RGB": "160,160,160",
                        "hex": "#A0A0A0",
                        "on": "#FFFFFF",
                      },
                      "tint80": Object {
                        "RGB": "105,105,105",
                        "hex": "#696969",
                        "on": "#FFFFFF",
                      },
                    },
                    "complimentary": Object {
                      "dark": Object {
                        "RGB": "206,185,0",
                        "hex": "#CEB900",
                        "on": "#272727",
                      },
                      "light": Object {
                        "RGB": "255,245,157",
                        "hex": "#FFF59D",
                        "on": "#272727",
                      },
                      "main": Object {
                        "RGB": "255,238,82",
                        "hex": "#FFEE52",
                        "on": "#272727",
                      },
                    },
                    "error": Object {
                      "dark": Object {
                        "RGB": "197,73,1",
                        "hex": "#C54901",
                        "on": "#FFFFFF",
                      },
                      "main": Object {
                        "RGB": "255,115,32",
                        "hex": "#FF7320",
                        "on": "#FFFFFF",
                      },
                    },
                    "primary": Object {
                      "dark": Object {
                        "RGB": "102,22,20",
                        "hex": "#661614",
                        "on": "#FFFFFF",
                      },
                      "light": Object {
                        "RGB": "218,52,47",
                        "hex": "#DA342F",
                        "on": "#FFFFFF",
                      },
                      "main": Object {
                        "RGB": "164,40,36",
                        "hex": "#A42824",
                        "on": "#FFFFFF",
                      },
                    },
                    "secondary": Object {
                      "dark": Object {
                        "RGB": "5,38,59",
                        "hex": "#05263B",
                        "on": "#FFFFFF",
                      },
                      "light": Object {
                        "RGB": "89,156,175",
                        "hex": "#599CAF",
                        "on": "#FFFFFF",
                      },
                      "main": Object {
                        "RGB": "11,70,107",
                        "hex": "#0B466B",
                        "on": "#FFFFFF",
                      },
                    },
                    "surface": Object {
                      "RGB": "255,255,255",
                      "hex": "#FFFFFF",
                      "on": "#272727",
                    },
                    "white": Object {
                      "RGB": "255,255,255",
                      "hex": "#FFFFFF",
                      "on": "#272727",
                    },
                  },
                },
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Box__Body-sc-1hhoqar-2",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "background-color:",
                    [Function],
                    ";border-radius:",
                    [Function],
                    ";border:1px solid ",
                    [Function],
                    ";",
                  ],
                },
                "displayName": "Body",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Box__Body-sc-1hhoqar-2",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            theme={
              Object {
                "action": Object {
                  "dark": Object {
                    "RGB": "79,38,98",
                    "hex": "#4E2662",
                    "on": "#FFFFFF",
                  },
                  "light": Object {
                    "RGB": "173,101,207",
                    "hex": "#AD65CF",
                    "on": "#FFFFFF",
                  },
                  "main": Object {
                    "RGB": "116,60,143",
                    "hex": "#743C8F",
                    "on": "#FFFFFF",
                  },
                },
                "background": Object {
                  "RGB": "246,246,246",
                  "hex": "#F6F6F6",
                },
                "black": Object {
                  "main": Object {
                    "RGB": "39,39,39",
                    "hex": "#272727",
                    "on": "#FFFFFF",
                  },
                  "tint20": Object {
                    "RGB": "236,236,236",
                    "hex": "#ECECEC",
                    "on": "#272727",
                  },
                  "tint40": Object {
                    "RGB": "199,199,199",
                    "hex": "#C7C7C7",
                    "on": "#272727",
                  },
                  "tint60": Object {
                    "RGB": "160,160,160",
                    "hex": "#A0A0A0",
                    "on": "#FFFFFF",
                  },
                  "tint80": Object {
                    "RGB": "105,105,105",
                    "hex": "#696969",
                    "on": "#FFFFFF",
                  },
                },
                "complimentary": Object {
                  "dark": Object {
                    "RGB": "206,185,0",
                    "hex": "#CEB900",
                    "on": "#272727",
                  },
                  "light": Object {
                    "RGB": "255,245,157",
                    "hex": "#FFF59D",
                    "on": "#272727",
                  },
                  "main": Object {
                    "RGB": "255,238,82",
                    "hex": "#FFEE52",
                    "on": "#272727",
                  },
                },
                "error": Object {
                  "dark": Object {
                    "RGB": "197,73,1",
                    "hex": "#C54901",
                    "on": "#FFFFFF",
                  },
                  "main": Object {
                    "RGB": "255,115,32",
                    "hex": "#FF7320",
                    "on": "#FFFFFF",
                  },
                },
                "primary": Object {
                  "dark": Object {
                    "RGB": "102,22,20",
                    "hex": "#661614",
                    "on": "#FFFFFF",
                  },
                  "light": Object {
                    "RGB": "218,52,47",
                    "hex": "#DA342F",
                    "on": "#FFFFFF",
                  },
                  "main": Object {
                    "RGB": "164,40,36",
                    "hex": "#A42824",
                    "on": "#FFFFFF",
                  },
                },
                "secondary": Object {
                  "dark": Object {
                    "RGB": "5,38,59",
                    "hex": "#05263B",
                    "on": "#FFFFFF",
                  },
                  "light": Object {
                    "RGB": "89,156,175",
                    "hex": "#599CAF",
                    "on": "#FFFFFF",
                  },
                  "main": Object {
                    "RGB": "11,70,107",
                    "hex": "#0B466B",
                    "on": "#FFFFFF",
                  },
                },
                "surface": Object {
                  "RGB": "255,255,255",
                  "hex": "#FFFFFF",
                  "on": "#272727",
                },
                "white": Object {
                  "RGB": "255,255,255",
                  "hex": "#FFFFFF",
                  "on": "#272727",
                },
              }
            }
          >
            <div
              className="c2"
            >
              <span />
            </div>
          </StyledComponent>
        </Body>
      </div>
    </StyledComponent>
  </StyledBox>
</Box>
`;

exports[`Box test should match last snapshot without a title 1`] = `
<StyledBox
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <Body
    flatTop={false}
    theme={
      Object {
        "action": Object {
          "dark": Object {
            "RGB": "79,38,98",
            "hex": "#4E2662",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "173,101,207",
            "hex": "#AD65CF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "116,60,143",
            "hex": "#743C8F",
            "on": "#FFFFFF",
          },
        },
        "background": Object {
          "RGB": "246,246,246",
          "hex": "#F6F6F6",
        },
        "black": Object {
          "main": Object {
            "RGB": "39,39,39",
            "hex": "#272727",
            "on": "#FFFFFF",
          },
          "tint20": Object {
            "RGB": "236,236,236",
            "hex": "#ECECEC",
            "on": "#272727",
          },
          "tint40": Object {
            "RGB": "199,199,199",
            "hex": "#C7C7C7",
            "on": "#272727",
          },
          "tint60": Object {
            "RGB": "160,160,160",
            "hex": "#A0A0A0",
            "on": "#FFFFFF",
          },
          "tint80": Object {
            "RGB": "105,105,105",
            "hex": "#696969",
            "on": "#FFFFFF",
          },
        },
        "complimentary": Object {
          "dark": Object {
            "RGB": "206,185,0",
            "hex": "#CEB900",
            "on": "#272727",
          },
          "light": Object {
            "RGB": "255,245,157",
            "hex": "#FFF59D",
            "on": "#272727",
          },
          "main": Object {
            "RGB": "255,238,82",
            "hex": "#FFEE52",
            "on": "#272727",
          },
        },
        "error": Object {
          "dark": Object {
            "RGB": "197,73,1",
            "hex": "#C54901",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "255,115,32",
            "hex": "#FF7320",
            "on": "#FFFFFF",
          },
        },
        "primary": Object {
          "dark": Object {
            "RGB": "102,22,20",
            "hex": "#661614",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "218,52,47",
            "hex": "#DA342F",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "164,40,36",
            "hex": "#A42824",
            "on": "#FFFFFF",
          },
        },
        "secondary": Object {
          "dark": Object {
            "RGB": "5,38,59",
            "hex": "#05263B",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "89,156,175",
            "hex": "#599CAF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "11,70,107",
            "hex": "#0B466B",
            "on": "#FFFFFF",
          },
        },
        "surface": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
        "white": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
      }
    }
  >
    <span>
      heleo
    </span>
  </Body>
</StyledBox>
`;
